main.go

```diff
@@ -367,15 +379,15 @@ func makeSecretChanges(kubeClient *k8s.Client, secret *corev1.Secret, initiator
 
 		// clean up acme challenge records afterwards
 		// for _, hostname := range hostnames {
-		// 	log.Info().Msgf("[%v] Secret %v.%v - Cleaning up TXT record _acme-challenge.%v...", initiator, *secret.Metadata.Name, *secret.Metadata.Namespace, hostname)
+		// 	log.Info().Msgf("[%v] Secret %v.%v - Cleaning up TXT record _acme-challenge.%v...", initiator, secret.Name, secret.Namespace, hostname)
 		// 	err = cloudflareProvider.CleanUp(hostname, "", "123d==")
 		// 	if err != nil {
-		// 		log.Info().Err(err).Msgf("[%v] Secret %v.%v - Cleaning up TXT record _acme-challenge.%v failed", initiator, *secret.Metadata.Name, *secret.Metadata.Namespace, hostname)
+		// 		log.Info().Err(err).Msgf("[%v] Secret %v.%v - Cleaning up TXT record _acme-challenge.%v failed", initiator, secret.Name, secret.Namespace, hostname)
 		// 	}
 		// }
 
 		// reload secret to avoid object has been modified error
-		err = kubeClient.Get(context.Background(), *secret.Metadata.Namespace, *secret.Metadata.Name, secret)
+		secret, err = kubeClientset.CoreV1().Secrets("").Get(secret.Name, metav1.GetOptions{})
```

Comment:
Plug in the `secret.Namespace` inside `.Secrets(...)` to fetch it from the correct namespace.